core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java

```diff
@@ -244,10 +247,15 @@ public void onFailure(Throwable e) {
         for (int i = 0; i < numDocs; i++) {
             try {
                 synchronized (indexVersionLock) {
-                    client().prepareIndex("test", "test").setSource("index_version", indexVersion.get()).get();
+                    client().prepareIndex("test", "test").setSource("index_version", indexVersion.get()).setTimeout(TimeValue.timeValueSeconds(10)).get();
                 }
             } catch (IndexNotFoundException inf) {
-                // fine
+                // fine index has been deleted
+            } catch (ProcessClusterEventTimeoutException ex) {
```

Comment:
Works for me... ( I presume you did see it fail due to this in practice - I did see it when setting timeout to 10ms, but that's just because index creation wasn't fast enough?). The other unavailable shards exception is a valid addition though. We should still add it , even if disabling the test.